game/tests.py

```
@@ -472,6 +472,52 @@ def test_capture_tracked(self):
         self.assertEqual(data['captured'], 'p')
 
 
+class ServerSideMoveValidationTest(TestCase):
+    """Test server-side move validation resists stale or tampered clients."""
+
+    def setUp(self):
+        self.client.get('/play/')
+
+    def _session_game_data(self):
```

Ruff (0.15.15)

[warning] Missing return type annotation for private function `_session_game_data`

(ANN202)